### ISSUE TITLE
Resizing from left or top edge or maximizing window introduces mouse offset (Linux only)

### ISSUE THREAD

**Issue by poli0048** [ocornut/imgui#2935]
Revision: aa8392c (2019-11-18 14:41:02)
Branch: Docking

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Compiler: GCC 8.3
Operating System: Linux

Issue Behavior: When resizing or after resizing a platform window from the left edge, top edge, or top-left corner, there is an offset between the actual location of the mouse cursor and where ImGui thinks the cursor is relative to its widgets. For instance, a button may not highlight anymore when the cursor is hovering over it, but it might highlight if the cursor is next to it instead. It actually looks like the locations and sizes of the widgets are updated on screen, but the areas where you can interact with them doesn't change. The behavior is the same if a platform window is maximized.

Issue Details: After some digging, it looks like the backend (imgui_impl_glfw.cpp) only triggers an update to the position of a viewport's origin in the GLFW window move callback (by setting viewport->PlatformRequestMove = true). In the GLFW window resize callback a viewport resize is triggered (by setting viewport->PlatformRequestResize = true), but the viewport's origin is not updated.

I have confirmed on Linux that with the latest version of GLFW (updated today) resizing a window from any edge or corner will trigger the ImGui_ImplGlfw_WindowSizeCallback but not the ImGui_ImplGlfw_WindowPosCallback. This would seem to explain the behavior, since a resize event from the left or top edge, or the top-left corner will change the location of the window origin, but not trigger a ImGui_ImplGlfw_WindowPosCallback to update the viewport location state. 

Additional details: Some more testing revealed that on Windows such a window resize triggers both of the above GLFW callbacks, which means that this issue does not occur on Windows. Since the GLFW behavior is not consistent across platforms it would be fair to characterize this as an ImGui issue, but a GLFW bug. That being said, it can be worked around pretty easily by setting "viewport->PlatformRequestMove = true" in ImGui_ImplGlfw_WindowSizeCallback(). I can confirm that this resolves the issue.

Minimal Example:
Just run the demo without modifications and resize from the left or top edge and you should see this behavior (if using GLFW backend on Linux).

**Comment by rokups** [comment]
I am unable to replicate the issue. Tested with multiple glfw versions including git version.
![glfw](https://user-images.githubusercontent.com/19151258/71154102-7045d780-2243-11ea-99eb-d6cfe09d5fbc.gif)

Also commit you linked is forked from master branch, not docking. I suggest you check out docking branch and test it.

**Comment by ocornut** [comment]
It probably would be useful for both of you to tell which Window Manager you are using.

**Comment by rokups** [comment]
Indeed. So many moving parts in linux ecosystem. I tested on kwin_x11.

**Comment by poli0048** [comment]
Good point. I'm on Debian Buster with Gnome 3.32 on X11 (Not Wayland).

rokups, thanks for trying this out. Can you put a printf statement in function ImGui_ImplGlfw_WindowPosCallback() in file imgui_impl_glfw.cpp, and verify whether or not you are getting this callback when you resize a window from the top, left, or top-left? I have confirmed that I definitely am on Windows, but in my Linux environment I definitely am not.

I have posted this issue on the GLFW project - if it happens on my setup but not yours that could be useful info for them and I would be happy to add that info there.

**Comment by rokups** [comment]
`ImGui_ImplGlfw_WindowPosCallback` does get called.

What about version you test? You linked commit that is unrelated to docking branch. It probably was a mistake. I am curious what version exactly you are  testing.

**Comment by poli0048** [comment]
I'm using a fork: https://github.com/malamanteau/imgui
The commit I listed was the most recent main branch commit I saw in the log. I am not great with Git - sorry if this was not the right piece of information.

Thanks for confirming the callback - I will add that info to the GLFW issue I made.

I guess the only thing to do here is to decide whether or not you want to depend on ImGui_ImplGlfw_WindowPosCallback() getting called on resize events or not. Looks like it usually is, and hopefully I can get GLFW team to make that behavior consistent everywhere. If you don't want to count on it, the addition in my original post seems like a reasonable change.

Thanks for the help with this. Best regards to both of you.

**Comment by rokups** [comment]
Could you clone stock docking branch from https://github.com/ocornut/imgui/ and test glfw sample if it has same behavior?

**Comment by poli0048** [comment]
Hi - sorry for the delay. I hope you had a nice holiday.

I did as you suggested and I cannot reproduce the behavior. I added a printf in the ImGui_ImplGlfw_WindowPosCallback() function and I can confirm that this callback *is* getting called when I resize a window from any edge or corner (if not captured by the main window). Not sure what could be changing this behavior, but it looks like this issue is specific to my fork. So in other words... not your problem :-)

Best regards,

Bryan
